requesttojson/requesttojson.go

```diff
@@ -0,0 +1,58 @@
+package requesttojson
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// Struct to represent the HTTP request in JSON format
+type RequestJSON struct {
+	Method  string              `json:"method"`
+	URL     string              `json:"url"`
+	Proto   string              `json:"proto"`
+	Headers map[string][]string `json:"headers"`
+	Body    string              `json:"body"`
+	Query   map[string][]string `json:"query_parameters"`
+}
+
+func RequestToJSON(req *http.Request, replaceEmptyBodyWith string) (json.RawMessage, RequestJSON, error) {
```

Comment:
wdyt of using `defaultBodyContent` instead `replaceEmptyBodyWith` ?
```suggestion
func RequestToJSON(req *http.Request, defaultBodyContent string) (json.RawMessage, RequestJSON, error) {
```